### ISSUE TITLE
Multiple service times

### ISSUE THREAD

**Issue by tim-pearce** [tbar0970/jethro-pmm#501]
We are not using Jethro for service stats yet but with our current stats system (a spreadsheet) we record attendance of individuals at each service (8:30 and 10:30) and also roll that up to an attendance for the day.
That is to say if a person attends one or both services it is marked as one attendance for the day and once against each respective service they have attended.
Currently Jethro only allows one service per day.
In my evaluation I played with having a group for each service, but that means EVERY member has to be part of each group - and what if we were to add a third service?!

**Comment by tbar0970** [comment]
So the enhancement here would be to make it possible to record attendance for the same congregation at several different times on the one date.  I guess you'd want to have selectable options for the service times.

How important is it to know exactly which individuals attended which service on a Sunday? 

Do you have basically the same set of people who are just as likely to attend either or both services?  Or do most people attend one or the other regularly?

**Comment by tim-pearce** [comment]
1. We normally have an 8:30 and a 10:30, but sometimes we have just one service

2. We have only had the two for a couple of years. There seems to be some movement dependent on the season. Mainly we want to know how many people would attend a single service. Just adding the two totals does not tell us that as there are people (myself included) who attend both services for various reasons.

**Comment by tbar0970** [comment]
My suggestion would be to have an 8.30 congregation and a 10.30 congregation in Jethro.  Put the people who typically come to 8.30 in the first, and the people who typically come to 10.30 in the second.  The fact that there's fluidity and double-ups is ok.

At your 8.30 service, open up the attendance record for *both* congregations and mark as present all the people you see there.  And, do a headcount and save that with the attendance in Jethro. Then at our 10.30 service, do the same (this time you can mark as "absent" everyone not already marked "present").  At the end of this process, you have a bunch of people marked present.  The "total present" in the 8.30 congregation plus the "total present" in the 10.30 congregation gives you the total number of distinct individuals who came to church that day. The headcounts for the two congregation tell you how many people were at each individual service.

**Comment by s4069b** [comment]
I just read that suggestion - I had not thought to do it that way. My work-around has been to create 'attendance groups' for each congregation. It means the attendance groups are larger (as the original poster pointed out) there's more 'maintenance' required, and we don't benefit from some of the features that are tied to congregational attendance. It is a bit clunky but gets the job done.  

The obvious down-side to your suggestion is that you don't have a record of who was in attendance at each service (you only have a head-count). It's actually quite helpful to know who was at which service because attendance at 'a different congregation' is not the same as attendance at your regular congregation. We want to encourage everyone to be active in their own congregation (which is also the way Jethro is geared). The head-count does not take 'age' into account either. So the attendance statistics are not as useful. Also there's bonuses like not needing to mark the attendance for kids church (because we can see who was there or not in the main attendance record). 

I've not fully thought this through but here's two alternative ideas:
- a third category of attendance. Present, Absent, ?, and 'present but at another service'
- re-work (significant re-work I'm guessing!) so that you can mark a person as being in attendance at any congregation in the system, and at more than one congregation in the system. The process of marking attendance may be to select the congregations (similar to the way we do now from drop-downs) and click 'tabular' (or a new box 'multiple') and mark attendance at each and ever congregation they showed up at.  Statistics would then need reworking too!

**Comment by tbar0970** [comment]
> attendance at 'a different congregation' is not the same as attendance at your regular congregation

Agree.  But what followup do you do based on someone's attendance at their not-normal congregation?  Or is it the overall stats you're interested in?

Related: Issue #490.  We could record explicitly which congregation somebody was present at.  Not sure if that would cover the multiple-attendance scenario.

**Comment by s4069b** [comment]
Here's some thoughts...

On Monday morning Jethro kindly emails a report to a group of elders/staff which includes anyone who has missed church for three weeks or more. I'd be less inclined to 'chase' someone who showed up at one of the other services.

We have had people 'swap' from one service to another. If we can see that they have been absent from 'their' congregation, but present at one of the others - then I know we need to have a conversation about their intentions and their commitment and service.  

On Monday morning Jethro also emails a report of 'visitors/newcomers' to our 'welcomers' and staff. We don't want to count someone as a visitor (for follow-up or for statistics) if they are from another congregation.

We need to report attendance stats to the denomination (they work out our contributions to the denomination based on attendance and financial records). They are only interested in 'adults' - so this is where a head-count is not as useful as the stats which show adult attendance. Double-counting can be an issue here (for someone who attends more than once).

**Comment by The-Big-V** [comment]
Hello friends,

I really appreciate everything your doing with Jethro. I hope this is right thread to put this comment.

A bit of background: we are using Jethro for our church in Melbourne, which is currently under COVID lockdown. We have two congregations: 10am and 5pm (we may start a third in the coming months).

The issue of recording attendance at multiple service is about to become a more pressing issue for us, because the Victorian government roadmap out of lockdown has indicated that the new "COVID normal" will **require religous organisations to keep a record of attendance at events**. This appears not to be a transitional situation, but a long-term one.

Ideally, we could have an iPad at the door and mark the roll as people arrive. However, at present we can only mark people present at one service. We would need to have a sign-in sheet to collect details of those who attend more than one service (we'll need that for newcomers anyway).

It would be great if we could:

- mark a roll of "all congregations"
- and then tick which services they were present at
- attendance reports could generate: _"10am attendance, 5pm attendance, unique attendees (the total of the two minus those who attended twice)."_

**Comment by s4069b** [comment]
I think you can come close to what you're after with a work-around. 
1. create a new Group for each service (tip. start the name with a zero so these groups are at the top of the drop-down eg" 0 9.30am service")
2. assign everyone who's likely to attend into each of the service groups which you just created - in this way you can include a person in more than one service group. 
3. Mark attendance for all service groups with the 'tabular' option ticked... so each person will have a ? A P against their name in each of the service 'group' columns. 
4. When you view results you can select tabular again, or total if you just want to see who attended at least one service. 

It's a pretty solid work-around which I've been using for a few years now.

**Comment by The-Big-V** [comment]
Brilliant, thanks.
